tests/setuponlytests/test.sh

```diff
@@ -46,7 +46,10 @@ setupOnlyMinecraftTest(){
     outputContainerLog "$logs"
     result=1
   elif [ -f verify.sh ]; then
-    if ! docker run --rm --entrypoint bash -v "${PWD}/data":/data -v "${PWD}/verify.sh":/verify "${IMAGE_TO_TEST:-itzg/minecraft-server}" -e /verify; then
+    # if there is an environment.sh, run it, and push the output to ./.tests.env. then load that as an env-file for the docker run
+    env_file=$([ -f environment.sh ] && sh ./environment.sh > ./.tests.env && echo "--env-file=./.tests.env" || echo "")
+    [ ! -z "${env_file// }" ] && echo "Using env-file argument: $env_file"
+    if ! docker run --rm --entrypoint bash $env_file -v "${PWD}/data":/data -v "${PWD}/verify.sh":/verify "${IMAGE_TO_TEST:-itzg/minecraft-server}" -e /verify; then
       endTime=$(date +%s)
```

Comment:
This all seems quite complicated and as previously implemented it was necessary. Looking closer at this I realized I should have been using `docker compose run` for the `verify` also. So, I went with an idea there and confirmed it works:

https://github.com/itzg/docker-minecraft-server/commit/5df07696f839cc87198b38a20993f7f234d43705#diff-f8a6ba079e5e45526e46f87838c9d21b0caff96739fc36d21e3b41ad767f77faR49

If you're good with this, I can just push my changes into your PR branch.

Reply:
Oh, so when you use `compose run` the variables are available? that's great. way less complicated. 👍 yes, lets go with that for sure.